mutable_tree.go

```diff
@@ -364,11 +366,37 @@ func (tree *MutableTree) GetVersioned(key []byte, version int64) (
 	return -1, nil
 }
 
+// SaveVersionMem saves a new tree version to disk, based on the current state of
+// the tree. Returns the hash and new version number.
+func (tree *MutableTree) SaveVersionMem() ([]byte, int64, error) {
+	return tree.saveVersion(false)
+}
+
+// FlushMemDisk saves a new tree to disk and removes all the versions in memory
+func (tree *MutableTree) FlushMemVersionDisk() ([]byte, int64, error) {
+	x, y, err := tree.saveVersion(true)
+	tree.ndb.dbMem = dbm.NewMemDB()
+	tree.memVersions = map[int64]bool{}
```

Comment:
IMHO, if we do not delete all memVersions and keep recent n version would be better, for it would fail when query recent version of state like querystore, if height is not exactly the latest one, it would fail for the version is just deleted.